Loop/Managers/DeviceDataManager.swift

```diff
@@ -1256,6 +1264,14 @@ extension DeviceDataManager {
             assertionFailure("\(#function) should be invoked only when a testing CGM manager is in use")
             return
         }
+        
+        guard !glucoseStore.sharingDenied else {
+            // only clear cache since access to health kit is denied
+            glucoseStore.purgeCachedGlucoseObjects() { error in
```

Comment:
Similar situation here, to the one described above.

Reply:
I think I’m missing something. For GlucoseStore purgeAllGlucoseSamplesit seems there are only 2 actions:
- purgeCachedGlucoseObjects()
- healthStore.deleteObjects
Is there an additional action related to CoreData that I’m overlooking?

Note: there is also a clean-up step handleUpdatedGlucoseData() but that step is also apart of purgeCachedGlucoseObjects()